ls8/cpu.c

```diff
@@ -103,16 +104,8 @@ void cpu_ram_write(struct cpu *cpu, unsigned char address, unsigned char value)
  */
 
 int num_operands_needed(unsigned char IR) {
-  switch (IR) {
-    case LDI:
-      return 2;
-
-    case PRN: 
-      return 1;
-
-    default:
-      return 0;
-  }
+  unsigned char AA = IR>>6;
```

Comment:
it makes sense to leave everything as `unsigned char` or 1 byte. The numbers go from 0-255, so you'll never need `int`